wikimedia/wikipedia-ios
# Add article inspector mapping (part 4)

**Fixes Phabricator ticket:** https://phabricator.wikimedia.org/T270426

Do not review until #3869 is merged and this PR is updated with `main`.

### Notes
* This completes the Phabricator task - it is the final step that adds editor and revision information to each`CombinedSentence` struct, as well as a way to generate `NSAttributedString`s for each `CombinedSentence`. 

### Test Steps

1. Set isEnabled = true in ArticleInspectorController
2. Add a breakpoint in `ArticleInspectorController` at the point that `self.combinedSections` is assigned.
3. Run and go to the article that is cached on the endpoint. You might need to visit https://wikiwho-ios-experiments.wmflabs.org/whocolor/{article_title} to kick off the caching. See that combinedSections are properly populated at the breakpoint with correct editor and revision information within each `CombinedSentence`'s `annotatedData` property (note I also added some light unit tests that we can flesh out more fully as we run into issues).

I also added the attributed string generation method as a helper on `CombinedSentence`. I figure we will call this only after we have chosen a sentence (that will be done in a later Phabricator task). IMO there's no reason to generate attributed strings for every sentence if we're only going to need them for one. Because of this there's no way yet to test this method in the app - for now you can set a breakpoint in the `testAttributedString` unit test method to inspect those strings that are generated
